container/go/pkg/compat/overrideConfig.go

```diff
@@ -0,0 +1,453 @@
+// if toFormat is set to True, adds a . in front of the value so that the string can be formatted.
+func keyValueToMap(value []string) map[string]string {
+	convMap := make(map[string]string)
+	var temp []string
+	for _, kvpair := range value {
+		temp = strings.Split(kvpair, "=")
+		key, val := temp[0], temp[1]
+		convMap[key] = val
+	}
+	return convMap
+}
+
+type formattedString map[string]interface{}
+
+// formateWithMap takes all variables of format {{.VAR}} in the input string `format`
+// and replaces it according to the map of parameters to values in `params`.
+func formatWithMap(format string, params formattedString) string {
+	msg := &bytes.Buffer{}
+	template.Must(template.New("").Parse(format)).Execute(msg, params)
+	return msg.String()
+}
+
+// Stamp provides the substitutions of variables inside {} using info in file pointed to
+// by stampInfoFile.
+func Stamp(inp string, stampInfoFile []string) (string, error) {
+	if len(stampInfoFile) == 0 || inp == "" {
+		return inp, nil
+	}
+	formatArgs := make(map[string]interface{})
+	for _, infofile := range stampInfoFile {
+		f, err := os.Open(infofile)
+		if err != nil {
+			return "", errors.Wrapf(err, "failed to open file %s", infofile)
+		}
+		defer f.Close()
+		// scanner reads line by line and discards '\n' character by default.
+		scanner := bufio.NewScanner(f)
+		var temp []string
+		for scanner.Scan() {
+			temp = strings.Split(scanner.Text(), " ")
+			key, val := temp[0], temp[1]
+			if _, ok := formatArgs[key]; ok {
+				fmt.Printf("WARNING: Duplicate value for key %s: using %s", key, val)
+			}
+			formatArgs[key] = val
+		}
+		if err = scanner.Err(); err != nil {
+			return "", errors.Wrapf(err, "failed to read line from file %s", infofile)
+		}
+	}
+	// do string manipulation in order to mimic python string format.
+	// specifically, replace '{' with '{{.' and '}' with '}}'.
+	inpReformatted := strings.ReplaceAll(inp, "{", "{{.")
+	inpReformatted = strings.ReplaceAll(inpReformatted, "}", "}}")
+	return formatWithMap(inpReformatted, formattedString(formatArgs)), nil
+}
+
+// mapToKeyValue reverses a map to a '='-separated array of strings in {key}={value} format.
+func mapToKeyValue(kvMap map[string]string) []string {
+	keyVals := []string{}
+	concatenated := ""
+	for k, v := range kvMap {
+		concatenated = k + "=" + v
+		keyVals = append(keyVals, concatenated)
+	}
+
+	sort.Sort(sort.StringSlice(keyVals))
+	return keyVals
+}
+
+// resolveVariables resolves the environment variables embedded in the given value using
+// provided map of environment variable expansions.
+// It handles formats like "PATH=$PATH:..."
+func resolveVariables(value string, environment map[string]string) (string, error) {
+	fmt.Printf("environment override %v", environment)
+	var i interface{}
+	i = "this is a string type"
+	var errorMet = false
+	mapper := func(p string) string {
+		switch i.(type) {
+		case string:
+			return environment[p]
+		}
+		errorMet = true
+		return ""
+	}
+	if errorMet {
```

Comment:
This will never be true here. This will only be true after `os.Expand` returns and a mapper invocation set `errorMet = true`. I suggest moving this check *after* os.Expand returns. Also, I would generate the error inside the mapper function and add the name of the bad key that was requested in the generated error.

Reply:
cannot generate error in mapper function because os.Expand only takes a function that doesn't return any errors.